ZipEditor/src/zipeditor/model/ZipModel.java

```diff
@@ -102,10 +106,24 @@ public static ContentTypeId detectType(InputStream contents) {
 			contents.reset();
 			if (count == -1)
 				return null;
-			ZipInputStream zip = new ZipInputStream(contents);
-			if (zip.getNextEntry() != null) {
-				contents.reset();
-				return ContentTypeId.ZIP_FILE;
+			try {
+				ZipArchiveInputStream zip = new ZipArchiveInputStream(contents) {
+					
+					@Override
+					protected InputStream createZstdInputStream(InputStream in) throws IOException {
+						return ZstdUtilities.getInputStream(in);
+					}
+					
+				};
+				if (zip.getNextEntry() != null) {
+					contents.reset();
+					return ContentTypeId.ZIP_FILE;
+				}
+			} catch (IOException ioe) {
+				if (ioe instanceof ZipEditorZstdException) {
+					return ContentTypeId.INVALID;
+				}
+				// thrown in getNextEntry() method if, if file is not a zip file.
```

Comment:
Not sure I understood this change & comment.
So if the exception is **not** `ZipEditorZstdException` we **silently continue**? Shouldn't we re-throw `IOException` here, or, better, catch `ZipEditorZstdException` only?